app/models/parser/twitter_profile.rb

```diff
@@ -19,25 +19,38 @@ def patterns
 
     # Main function for class
     def parse_data_for_parser(doc, _original_url, _jsonld_array)
-      @url = replace_subdomain_pattern(url)
-      username = url.match(/^https?:\/\/(www\.)?twitter\.com\/([^\/]+)$/)[2]
+      @url.gsub!(/\s/, '')
+      @url = replace_subdomain_pattern(url)      
+      username = compare_patterns(@url, self.patterns, 'username')
 
-      @parsed_data[:raw][:api] = {}
-      handle_twitter_exceptions do
-        @parsed_data[:raw][:api] = twitter_client.user(username).as_json
-        picture_url = parsed_data[:raw][:api][:profile_image_url_https].gsub('_normal', '')
-        set_data_field('picture', picture_url)
-        set_data_field('author_picture', picture_url)
+      @parsed_data[:raw][:api] = user_lookup_by_username(username)
+      
+      @parsed_data[:error] = parsed_data['raw']['api']['errors']
+      
+      if @parsed_data[:error] 
+        picture = ''
+        author_name = username
+        description = ''
+        published_at = ''
+      elsif @parsed_data[:error].nil?
+        raw_data = parsed_data['raw']['api']['data'][0]
+
+        picture = raw_data['profile_image_url'].gsub('_normal', '')
+        author_name = raw_data['name']
+        description = raw_data['description'].squish
+        published_at = raw_data['created_at']
       end
-      @parsed_data[:error] = parsed_data.dig(:raw, :api, :error)
-      set_data_field('title', parsed_data.dig(:raw, :api, :name), username)
+ 
       @parsed_data.merge!({
```

Comment:
**suggestion** same comments for this as in twitter item above. it's just a small thing, but think it'll help with readability since it'll be less code!